src/Raven.Server/Web/Studio/Processors/AbstractStudioCollectionsHandlerProcessorForPreviewRevisions.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Handlers.Processors;
+using Sparrow.Json;
+
+namespace Raven.Server.Web.Studio.Processors;
+
+internal abstract class AbstractStudioCollectionsHandlerProcessorForPreviewRevisions<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler>
+    where TOperationContext : JsonOperationContext
+    where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+{
+
+    protected readonly JsonContextPoolBase<TOperationContext> ContextPool;
+
+    protected string Collection;
+
+    protected AbstractStudioCollectionsHandlerProcessorForPreviewRevisions([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+    {
+        ContextPool = RequestHandler.ContextPool;
+    }
+
+    public override async ValueTask ExecuteAsync()
+    {
+        using (ContextPool.AllocateOperationContext(out TOperationContext context))
+        using (OpenReadTransaction(context))
+        using (var token = RequestHandler.CreateHttpRequestBoundOperationToken())
+        {
+            Collection = RequestHandler.GetStringQueryString("collection", required: false);
+
+            if (NotModified(context, out var etag))
```

Comment:
how can this be before `InitializeAsync` ? how would the sharded database know it is not modified?

Reply:
@shaharhikri ?

Reply:
Because if id isn't modified we want to return without doing anything, including 'InitializeAsync'

Reply:
but in sharded database how would you know without asking other shards first?

Reply:
I think the flow should be similar as in `AbstractStudioCollectionsHandlerProcessorForPreviewCollection` unless we have a bug there :)

Reply:
You right..
Done